File: DSA/Animesh_jha_590016813/day_36_question_1.c
#include <stdio.h>
#include <string.h>
#define MAX 100

char stack[MAX];
int top = -1;

void push(char c) {
    if (top < MAX - 1) {
        stack[++top] = c;
    }
}

char pop() {
    if (top >= 0) {
        return stack[top--];
    }
    return '\0';
}

void reverseString(char str[]) {
    int i;
    for (i = 0; str[i] != '\0'; i++) {
        push(str[i]);
    }

    for (i = 0; str[i] != '\0'; i++) {
        str[i] = pop();
    }
}

int main() {
    char str[100];

    printf("Enter a string: ");
    scanf("%s", str);  

    reverseString(str);

    printf("Reversed string: %s\n", str);
    return 0;
}

File: DSA/VaniAgrawal_590015846/Day_36_Question_1.py
# Utilize a stack's LIFO property to reverse string characters by pushing all characters and then popping them sequentially. 
def reverse_string():
    s = input("Enter a string: ")
    stack = []
    for ch in s:
        stack.append(ch)
    
    reversed_str = ''
    while stack:
        reversed_str += stack.pop()
    
    print("Reversed string:", reversed_str)

reverse_string()
